test/datastore/query.js

```diff
@@ -25,116 +25,333 @@ var Query = require('../../lib/datastore/query.js');
+      var query = new Query(['kind1']);
+      assert.strictEqual(query.autoPaginateVal, true);
+    });
 
-  it('should default autoPaginate to false', function() {
-    var query = new Query(['kind1']);
-    assert.strictEqual(query.autoPaginateVal, true);
   });
 
-  it('should default autoPaginate() to true', function() {
-    var query = new Query(['kind1'])
+  describe('autoPaginate', function() {
+
+    it('should enable auto pagination', function() {
+      var query = new Query(['kind1'])
         .autoPaginate();
-    assert.strictEqual(query.autoPaginateVal, true);
-  });
 
-  it('should support setting autoPaginate to false', function() {
-    var query = new Query(['kind1'])
+      assert.strictEqual(query.autoPaginateVal, true);
+    });
+
+    it('should disable auto pagination when false is passed in', function() {
+      var query = new Query(['kind1'])
         .autoPaginate(false);
-    assert.strictEqual(query.autoPaginateVal, false);
+
+      assert.strictEqual(query.autoPaginateVal, false);
+    });
+
+    it('should not disable auto pagination with falsy values', function() {
+      var query = new Query(['kind1'])
+        .autoPaginate(null);
+
+      assert.strictEqual(query.autoPaginateVal, true);
+    });
+
+    it('should return a new query', function() {
+      var query = new Query(['kind1']);
+      var newQuery = query.autoPaginate(false);
+
+      assert.notEqual(query.autoPaginateVal, newQuery.autoPaginateVal);
+    });
+
   });
 
-  it('should support field selection by field name', function() {
-    var query = new Query(['kind1'])
-        .select(['name', 'title']);
-    assert.equal(query.selectVal[0], 'name');
-    assert.equal(query.selectVal[1], 'title');
+  describe('filter', function() {
+
+    it('should support filtering', function() {
+      var now = new Date();
+      var query = new Query(['kind1'])
+        .filter('date <=', now);
+      var filter = query.filters[0];
+
+      assert.equal(filter.name, 'date');
+      assert.equal(filter.op, '<=');
+      assert.equal(filter.val, now);
+    });
+
+    it('should support multiple filters', function() {
+      var now = new Date();
+      var query = new Query(['kind1'])
+          .filter('date <=', now)
+          .filter('name =', 'Title')
+          .filter('count >', 20);
+
+      assert.equal(query.filters[0].name, 'date');
+      assert.equal(query.filters[0].op, '<=');
+      assert.strictEqual(query.filters[0].val, now);
+
+      assert.equal(query.filters[1].name, 'name');
+      assert.equal(query.filters[1].op, '=');
+      assert.equal(query.filters[1].val, 'Title');
+
+      assert.equal(query.filters[2].name, 'count');
+      assert.equal(query.filters[2].op, '>');
+      assert.strictEqual(query.filters[2].val, 20);
+    });
+
+    it('should remove any whitespace surrounding the filter name', function() {
+      var query = new Query(['kind1'])
+        .filter('   count    >', 123);
```

Comment:
Spacing of `   count    >` here and the next test is a little weird. Was this intentional?